### PR TITLE
Add prepend option for Net::HTTP instrumentation

### PR DESCRIPTION
**Pull request by tannalynn**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
This change adds a new config option to enable the instrumentation of Net::HTTP to use prepend rather than monkey patching. 

# Related Github Issue
https://github.com/newrelic/newrelic-ruby-agent/issues/347

# Testing
Added a new test suite `net_http_prepend`. I pulled out all of the existing work in net_http test suite into a helper module and included that in both `net_http_prepend` and `net_http`. These needed to be in separate suites because the instrumentation is set up when the agent starts, and cannot switch instrumentation implementations while running.

### CODE REVIEW

**Review by mwlang — CHANGES_REQUESTED** [review]
Let's make prepend the default so we're accelerating adoption of this technique.

**Comment by tannalynn** [comment]
> Let's make prepend the default so we're accelerating adoption of this technique.

Sounds great, this is now done.

File: lib/new_relic/agent/configuration/default_source.rb [hunk 1]
```diff
@@ -830,6 +830,13 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :allowed_from_server => false,
           :description => 'If <code>true</code>, uses Module.prepend rather than alias_method for ActiveRecord instrumentation.'
         },
+        :prepend_net_instrumentation => {
+          :default => false,
```
Comment (mwlang) [inline lib/new_relic/agent/configuration/default_source.rb]: still false here?

**Review by mwlang — APPROVED** [review]
All Good.  Friday the 13th, notwithstanding. Let's Roll.

**Pull request merged**